core/src/main/java/org/wso2/testgrid/core/command/CommandHandler.java

```diff
@@ -56,6 +56,8 @@
                                      impl = GenerateTestPlanCommand.class),
                          @SubCommand(name = "run-testplan",
                                      impl = RunTestPlanCommand.class),
+                         @SubCommand(name = "resolve-statuses",
+                                     impl = ResolveStatusesCommand.class),
```

Comment:
We need to call this command from the Jenkins pipeline exception path, right?

Reply:
We have added it to the post block. Which one would be better?

Reply:
But, then we have to wait until all testplans to complete before this is run. In the meanwhile, the Dashboard will be in an erroneous state.

I think each stage is already aware of its testplan-id. It is present in the test-plan yaml. Since we do a DB update at generate-testplans phase, the testplan yaml has the testplan-id. We can pass the entire testplan file as an input to finalize-run-testplan.

WDYT?